packages/mask/src/plugins/Savings/hooks/useYearnTokens.ts

```diff
@@ -0,0 +1,53 @@
+import { useAsync } from 'react-use'
+
+import { ChainId, EthereumTokenType, FungibleTokenDetailed, useFungibleTokensDetailed } from '@masknet/web3-shared-evm'
+import { useState } from 'react'
+import { VaultInterface, Yearn } from '@yfi/sdk'
+
+import { isValidYearnChain, splitToPair } from '../utils'
+import { flatten, compact, orderBy, sortedUniqBy } from 'lodash-unified'
+import type Web3 from 'web3'
+
+export function useYearnTokens(chainId: ChainId, web3: Web3) {
+    const [tokenPairs, setTokenPairs] = useState<[FungibleTokenDetailed, FungibleTokenDetailed][]>([])
+
+    const { value: yfiTokens, loading } = useAsync(async () => {
+        if (!isValidYearnChain(chainId)) {
+            return []
+        }
+
+        // @ts-ignore
```

Comment:
plz add reason for all `@ts-ignore` comments.

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-6.html#suppress-errors-in-ts-files-using--ts-ignore-comments